### ISSUE TITLE
`solr.ping()` returns 404: `SolrError: Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]`

### ISSUE THREAD

**Issue by ljmartin** [django-haystack/pysolr#380]
Hi PySolr - thanks for the library. Ran into a problem running the example script in the README.md. Any comments are appreciated, thanks!

# I have
* [x] Tested with the latest release
* [ ] Tested with the current master branch
* [x] Searched for similar existing issues

## Expected behaviour
Run the example script in the README without error

## Actual behaviour
`solr.ping()` returns 
```SolrError: Solr responded with an error (HTTP 404): [Reason: Error 404 Not Found]```

## Steps to reproduce the behaviour

1. Instantiate a solr instance using the tutorial at https://solr.apache.org/guide/8_11/solr-tutorial.html
2. open a python shell
3.
```python
import pysolr
# Create a client instance. The timeout and authentication options are not required.
solr = pysolr.Solr('http://localhost:8983/solr/')
solr.ping()
```
4. returns error above
5. Ensure that the solr instance is running on port 8983 ✅ 

## Configuration

* Operating system version: MacOS Big Sur 11.2.3
* Search engine version:
```

    solr-spec
        8.11.1

    solr-impl
        8.11.1 0b002b11819df70783e83ef36b42ed1223c14b50 - janhoy - 2021-12-14 13:50:55

    lucene-spec
        8.11.1

    lucene-impl
        8.11.1 0b002b11819df70783e83ef36b42ed1223c14b50 - janhoy - 2021-12-14 13:46:43

```

* Python version: `Python 3.9.7`
* pysolr version: 3.9.0 fresh install via mamba

**Comment by ljmartin** [comment]
Sorry about that - I didn't write the index name i.e. 
```
import pysolr
# Create a client instance. The timeout and authentication options are not required.
solr = pysolr.Solr('http://localhost:8983/solr/techproducts/')
solr.ping()
```

**Comment by bkline** [comment]
If what you originally reported doing is always going to fail, why does the documentation tell you to do it that way (see the repository's README)?

**Comment by cclauss** [comment]
@bkline Do you want to create a pull request to modify the README to match the new best practice?

**Comment by bkline** [comment]
Sure, as soon as I figure out what it should say. I've never used Solr before, so I've got some research to do. I just stood up a new docker container and tried to follow the instruction on the README, and I got the same error the O.P. got. Did `solr.ping()` previously work before any indices had been created? (I would assume so, given the implications of "_new_ best practice").

**Comment by bkline** [comment]
Are there release notes which discuss this change in behavior?

**Comment by cclauss** [comment]
The changelog is:
* https://github.com/django-haystack/pysolr/blob/master/CHANGELOG.rst and
* https://github.com/django-haystack/pysolr/releases

However the [last release to PyPI](https://pypi.org/project/pysolr/) was in 2020 and this issue was created in 2022, so I can find no release notes that discuss it.  That is why your documentation update would be really helpful.

**Comment by bkline** [comment]
Thanks. Still struggling with trying to get the package to work at all. I stumbled across a [tutorial](https://medium.com/@giovani.angelo/building-a-python-application-with-apache-solr-a-comprehensive-crud-tutorial-e8bf4e446d0f) which seemed as if it was going to show me how to get past the initial problem of not having an index, but that raised more questions than answers. To begin with, it appears to be working under the same assumption as the README we're trying to fix (that `ping()` will work with a new Solr instance which has no indices). But the showstopper was that it used a `create_core()` call which got the error message

```
AttributeError: 'Solr' object has no attribute 'create_core'
```

I feel a little bit as if I've half-stumbled into a parallel universe, where everyone else is writing about a different package than the one I installed. Perhaps I've overlooked something really basic, and the confusion is all of my own creation. It's happened before. 😅
